### ISSUE TITLE
Inception Resnet V2 model not classifying as expected

### ISSUE THREAD

**Issue by deepakrox** [tensorflow/models#429]
## Please let us know which model this issue is about (specify the top-level directory)
slims/nets/inception_resnet_v2.py

Hi, 
I am trying to build the inception_resnet_v2 model and restore the weights with the checkpoint file released (http://download.tensorflow.org/models/inception_resnet_v2_2016_08_30.tar.gz). I am using the following code to build, restore and classify using the model-

```
import tensorflow as tf
slim = tf.contrib.slim
from PIL import Image
from inception_resnet_v2 import *
import numpy as np

checkpoint_file = 'inception_resnet_v2_2016_08_30.ckpt'
sample_images = ['dog.jpg', 'panda.jpg']
#Load the model
sess = tf.Session()
arg_scope = inception_resnet_v2_arg_scope()
with slim.arg_scope(arg_scope):
  logits, end_points = inception_resnet_v2(input_tensor, is_training=False)
saver = tf.train.Saver()
saver.restore(sess, checkpoint_file)
for image in sample_images:
  im = Image.open(image).resize((299,299))
  im = np.array(im)
  im = im.reshape(-1,299,299,3)
  predict_values, logit_values = sess.run([end_points['Predictions'], logits], feed_dict={input_tensor: im})
  print (np.max(predict_values), np.max(logit_values))
  print (np.argmax(predict_values), np.argmax(logit_values))
```

It classifies the dog and panda incorrectly as belonging to the same(class number 918) with very very high confidence(~100%).

**Comment by jmchen-g** [comment]
Is it just for the two examples or all classes are messed up? Could it be just the model is not good at this specific class?

**Comment by deepak-matroid** [comment]
No.It seems to do this for all images I tried. Just put the two as examples. Tried it for other imagenet classes and it always seems to return the same class.

**Comment by deepak-matroid** [comment]
I figured out the issue. The image needs to be pre-processed in a certain way for the model to classify. Refer to http://stackoverflow.com/questions/39582703/using-pre-trained-inception-resnet-v2-with-tensorflow/39597537#39597537 for more info.

**Comment by deepakrox** [comment]
Figured out the issue - The image needs to be pre-processed in a certain way for the model to work. Interested folks can refer to http://stackoverflow.com/questions/39582703/using-pre-trained-inception-resnet-v2-with-tensorflow/39597537#39597537 for more details. Since this resolves the problem, this issue can be closed.
Thanks

**Comment by flyfj** [comment]
@deepakrox could you share the complete code? I have similar issue with VGG model, have applied preprocessing but the results are still not right. thanks!

**Comment by deepakrox** [comment]
@flyfj Working code attached
```
import tensorflow as tf
slim = tf.contrib.slim
from PIL import Image
from inception_resnet_v2 import *
import numpy as np

checkpoint_file = 'inception_resnet_v2_2016_08_30.ckpt'
sample_images = ['dog.jpg', 'panda.jpg']

input_tensor = tf.placeholder(tf.float32, shape=(None,299,299,3), name='input_image')
scaled_input_tensor = tf.scalar_mul((1.0/255), input_tensor)
scaled_input_tensor = tf.sub(scaled_input_tensor, 0.5)
scaled_input_tensor = tf.mul(scaled_input_tensor, 2.0)

#Load the model
sess = tf.Session()
arg_scope = inception_resnet_v2_arg_scope()
with slim.arg_scope(arg_scope):
  logits, end_points = inception_resnet_v2(scaled_input_tensor, is_training=False)
saver = tf.train.Saver()
saver.restore(sess, checkpoint_file)
for image in sample_images:
  im = Image.open(image).resize((299,299))
  im = np.array(im)
  im = im.reshape(-1,299,299,3)
  predict_values, logit_values = sess.run([end_points['Predictions'], logits], feed_dict={input_tensor: im})
  print (np.max(predict_values), np.max(logit_values))
  print (np.argmax(predict_values), np.argmax(logit_values))
```

**Comment by D-X-Y** [comment]
@deepakrox Hi, I use your code to classify "cropped_panda.jpg" used in tutorials/image/imagenet/classify_image.py.

But it seems output the wrong class (0.83959323, 8.7706404) (389, 389)

I also try a dog picture downloaded from web, and get (0.85302502, 9.4753761) (209, 209).

Would you mind help me figure where did I miss (I only change sample_images)?

Additionally, do your know the difference between 
```
image_data = tf.gfile.FastGFile(image, 'r').read()
decode_jpeg_data = tf.placeholder(tf.string)
decode_jpeg = tf.image.decode_jpeg(self._decode_jpeg_data, channels=3)
feed_dict={decode_jpeg_data: image_data}
```
and
```
im = Image.open(image).resize((299,299))
im = np.array(im)
im = im.reshape(-1,299,299,3)
```
they are all image reader, but get the difference output through the network.

Thanks!

**Comment by ayushidalmia** [comment]
@D-X-Y, where do we need to make the changes?
PS: I am new to this.

**Comment by ayushidalmia** [comment]
@deepakrox @D-X-Y 
Can we directly use the following for preprocessing:

```
for image in sample_images:
   decoded_jpeg =  tf.image.decode_jpeg(image)
    im = preprcess_for_train(decoded_jpeg,299,299)
   predict_values, logit_values = sess.run([end_points['Predictions'], logits], feed_dict={input_tensor: im})
   .....  #code as given by @deepakrox 

```

**Comment by auzyze** [comment]
@D-X-Y
Hi, I tried the same code with panda image, and also got 389, but that is not the correct label, is it? I think 169 should be the correct result ... 

I'm struggling to get correct prediction with slim pretrained model, tried different models and different preprocessing, but always get 389, which I think is the wrong result, anyone has comments on this? Thanks.

**Comment by zhejiangyyf** [comment]
I got similar problem. The model give the same result with high confidence for different images.
But in my case,I foget to add biases to the "up":
up=tf.nn.bias_add(up,bias)
The problem disappeared after correct the issue.
